File: backend/typescript/services/interfaces/emailService.ts
interface IEmailService {
  /**
   * Send email
   * @param from sender's email, formatted as '"Display Name" <address@email.com>'
   * @param to recipient's email
   * @param subject email subject
   * @param htmlBody email body as html
   * @throws Error if email was not sent successfully
   */
  sendEmail(
    from: string,
    to: string,
    subject: string,
    htmlBody: string,
  ): Promise<void>;
}

export default IEmailService;

File: backend/python/app/services/interfaces/email_service.py
from abc import ABC, abstractmethod


class IEmailService(ABC):
    """
    EmailService interface for handling email related functionality
    """

    @abstractmethod
    def send_email(self, to, subject, body):
        """
        Sends email with given parameters

        :param to: recipient's email
        :type to: str
        :param subject: email subject
        :type subject: str
        :param body: email body as html
        :type body: str
        :return: a dict that contains fields like id, threadId, labelIds
        of the sent email
        :rtype: dict
        :raises Exception: if email was not sent successfully
        """
        pass
